middleware/indexer.js

```diff
@@ -13,24 +12,37 @@ async function getPgClient() {
     return client;
 }
 
-async function findAccountsByPublicKeyIndexer(ctx) {
-    const { publicKey } = ctx.params;
+async function getStakingTxs(ctx) {
+    const { accountId } = ctx.params;
 
     const client = await getPgClient();
-    const { rows } = await client.query('SELECT DISTINCT account_id FROM access_keys WHERE public_key = $1', [publicKey]);
-    ctx.body = rows.map(({ account_id }) => account_id);
+    const { rows } = await client.query(`
+        select 
+        receiver_account_id validator_id,
+        args->>'method_name' method_name,
+        coalesce(nullif(args->>'deposit', '0'), convert_from(decode(args->>'args_base64', 'base64'), 'utf8')::jsonb->>'amount', '0') amount,
+        included_in_block_timestamp ts
```

Comment:
should the query also order in descending order by timestamp so that when it hits `limit` – we pick the most recent records?

Reply:
Consider someone stakes with one validator, does 9999 txs with other validators, then comes back and unstakes a portion from the original validator.

The wallet would have no clue how to calc rewards.

Reply:
We are making this query over network and handle result in JS. It's not going to work fast with 10k records. Postgres going to be much faster sorting and throwing away unneeded data than client-side.

> Consider someone stakes with one validator, does 9999 txs with other validators, then comes back and unstakes a portion from the original validator.

Sounds like we need to group by validator then? What do we need to get logically – last transaction with every validator?

Reply:
Each staking tx is only a piece of the overall deposit puzzle. Ideally, we need them all to be sure that we're not missing some amount user deposited with validator long ago. We could feed in list of all validators to narrow scope of txs down.